src/Symfony/Component/Console/Command/Command.php

```diff
@@ -648,6 +654,29 @@ public function getHelper(string $name): HelperInterface
         return $this->helperSet->get($name);
     }
 
+    private function writeDeprecationMessages(InputDefinition $inputDefinition, InputInterface $input, OutputInterface $output): void
+    {
+        $deprecationMessages = [];
+        foreach ($inputDefinition->getOptions() as $inputOption) {
+            if ($inputOption->isDeprecated()) {
+                try {
+                    $optionName = $inputOption->getName();
+                    $optionValue = $input->getOption($optionName);
+                    if (isset($optionValue) && $optionValue !== $inputOption->getDefault()) {
```

Comment:
> It may take a deeper work to detect if an option has been passed. This could be a feature in its own.

As you have already written, it is currently not possible to recognise whether an option has been used or whether the default value is simply being used.

Hence the check against the default value

Reply:
You can use `$input->hasParameterOption()` to check if an option is provided:

```php
$names = ['--'.$inputOption->getName()];
if ($inputOption->getShortcut()) {
    $names[] = '-'.$inputOption->getShortcut();
}
if ($input->hasParameterOption($names, true)) {
    $deprecationMessages[] = sprintf('The option "%s" is deprecated.', implode('|', $names));
}
```

Reply:
@GromNaN's suggested change applied